.circleci/config.yml

```
@@ -85,6 +85,19 @@ jobs:
       - run:
           name: Run pylint
           command: ./tools/pylint.sh
+  ruff:
+    docker:
+      - image: cimg/python:3.11.7
+    resource_class: small
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run:
```

You probably want to also ensure the venv where ruff was installed to in *pip-install* is activated so the command can be found
```suggestion
      - run:
          name: Enable virtual environment
          command: echo "source .venv/bin/activate" >> $BASH_ENV
      - run:
```

---

Ruff is written is Rust, not Python so it does not need the venv activated.